pkg/reconciler/reconciler.go

```diff
@@ -439,6 +440,18 @@ func WithUninstallAnnotations(as ...annotation.Uninstall) Option {
 	}
 }
 
+// WithPauseReconcileAnnotation is an Option that sets
+// a PauseReconcile annotation. If the Custom Resource watched by this
+// reconciler has the given annotation, and its value is set to `true`,
+// then reconciliation for this CR will not be performed until this annotation
+// is removed.
+func WithPauseReconcileAnnotation(annotationName string) Option {
+	return func(r *Reconciler) error {
+		r.pauseReconcileAnnotation = annotationName
+		return nil
+	}
+}
```

Comment:
I know there is some precedent already for annotation-based APIs in helm-operator-plugins, but I wonder if we could change this to something like the following to be more generic and make it possible to have a first-class spec field or even arbitrary logic to decide when to pause (but also still support the annotation-based API)

```go
func WithPauseReconcileHandler(handler func(context.Context, obj *unstructured.Unstructured) (bool, error)) {
	return func(r *Reconciler) error {
		r.pauseHandler = handler
		return nil
	}
}

Reply:
Done, PTAL